pkg/basecluster/prepare.go

```diff
@@ -68,19 +74,31 @@ func Prepare(fileName string, validateOnly bool) (clusterName string, modifiedYa
 
 // -----------------------------------------------------------------------------
 
-func removeNamespaceThenEncode(obj runtime.Object, enc *yaml.Encoder) (modified bool, err error) {
+func removeNamespaceThenEncode(obj runtime.Object, enc *yaml.Encoder, casMax, casMin int) (modified bool, err error) {
 	log := logpkg.GetLogger()
 	unstr := &unstructured.Unstructured{}
 	if err := scheme.Scheme.Convert(obj, unstr, nil); err != nil {
 		return false, fmt.Errorf("failed to convert object: %s", err)
 	}
+	kind := unstr.GetKind()
 	ns := unstr.GetNamespace()
 	if ns != "" {
 		log.V(1).Info("removing namespace",
 			"resource", unstr.GetName(), "namespace", ns)
 		unstr.SetNamespace("")
 		modified = true
 	}
+	if kind == "MachineDeployment" {
+		annotations := unstr.GetAnnotations()
+		if annotations == nil {
+			annotations = map[string]string{}
+		}
+
+		annotations[casMaxAnnotationMachineDeployments] = strconv.Itoa(casMax)
```

Comment:
I was thinking: if the user's manifest already has these annotations defined, shouldn't we leave them alone?

Reply:
will do